hvsock.go

```diff
@@ -314,3 +316,33 @@ func (conn *HvsockConn) SetReadDeadline(t time.Time) error {
 func (conn *HvsockConn) SetWriteDeadline(t time.Time) error {
 	return conn.sock.SetWriteDeadline(t)
 }
+
+// HvSockRegisterService registers the application defined by the guid and name with the
+// Hyper-V Host's registry.
+//
+// See: https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/make-integration-service#register-a-new-application
+func HvSockRegisterService(id guid.GUID, name string) error {
+	k, exists, err := registry.CreateKey(registry.LOCAL_MACHINE, hvSocketRegKey(id), registry.WRITE)
+	if err != nil {
+		return fmt.Errorf("could not create registry for %s: %w", id.String(), err)
+	}
+	defer k.Close()
+
+	if exists {
+		return nil
```

Comment:
If it already exists should we change the name (or check the name and error if different) instead of returning?